File: VisaCheckout/src/main/java/com/braintreepayments/api/VisaCheckoutTokenizeCallback.java
package com.braintreepayments.api;

import com.visa.checkout.VisaPaymentSummary;

/**
 * Callback for receiving result of
 * {@link VisaCheckoutClient#tokenize(VisaPaymentSummary, VisaCheckoutTokenizeCallback)}.
 */
public interface VisaCheckoutTokenizeCallback {

    /**
     * @param paymentMethodNonce {@link BraintreeNonce}
     * @param error an exception that occurred while tokenizing a Visa payment method
     */
    void onResult(BraintreeNonce paymentMethodNonce, Exception error);
}

File: VisaCheckout/src/main/java/com/braintreepayments/api/VisaCheckoutTokenizeCallback.kt
package com.braintreepayments.api

/**
 * Callback for receiving result of [VisaCheckoutClient.tokenize].
 */
fun interface VisaCheckoutTokenizeCallback {
    /**
     * @param visaCheckoutResult a [VisaCheckoutResult]
     */
    fun onVisaCheckoutResult(visaCheckoutResult: VisaCheckoutResult)
}